packages/notion/utils.ts

```diff
@@ -95,3 +98,55 @@ function _selectHandler(propertyValue: Select): string {
   }
   return '';
 }
+
+export function extractResultElection(
+  blockList: ListBlockChildrenResponse['results'],
+  electionPageId: string,
+): Promise<ResultElectionData[]> {
+  const blocks = blockList.filter(
+    (block) => 'type' in block && block.type === 'heading_1',
+  );
+
+  const partiesResult: string[] = [];
+  const votes: number[] = [];
+
+  for (let block of blocks) {
+    if ('type' in block && block?.type === 'heading_1') {
+      const result = block?.heading_1?.rich_text[0].plain_text.split(' - ');
+      partiesResult.push(result[1]);
+      votes.push(Number(result[2].split(' ')[0]));
+    }
+  }
+
+  return _mapToResultPage(partiesResult, votes, electionPageId);
+}
+
+async function _mapToResultPage(
+  partiesResult: string[],
+  votes: number[],
+  electionPageId: string,
+): Promise<ResultElectionData[]> {
+  const resultParty = await getElectionPage(electionPageId);
+  const results: ResultElectionData[] = [];
+  const totalVotes = votes.reduce((a, b) => a + b, 0);
+  const percentage: string[] = [];
+
+  votes.map((x) => {
+    const result: number = (x * 100) / totalVotes;
+    percentage.push(result.toFixed(2));
+  });
+
+  partiesResult.map(function (value, index) {
+    const currentParty = resultParty.filter((party) => party.name === value);
+
+    results.push({
+      candidate: currentParty[0].members.candidate.name,
```

Comment:
```suggestion
    const [currentParty] = resultParty.filter((party) => party.name === value);

    results.push({
      candidate: currentParty.members.candidate.name,
```
da para simplificar levemente aq

Reply:
Boa ideia @vzsoares! Melhoria aplicada.